cmd/stat.go

```diff
@@ -0,0 +1,222 @@
+/*
+ * Minio Client (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	humanize "github.com/dustin/go-humanize"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/mc/pkg/probe"
+)
+
+// contentMessage container for content message structure.
+type statMessage struct {
+	Status            string            `json:"status"`
+	Key               string            `json:"name"`
+	Date              time.Time         `json:"lastModified"`
+	Size              int64             `json:"size"`
+	ETag              string            `json:"etag"`
+	Type              string            `json:"type"`
+	EncryptionHeaders map[string]string `json:"encryption"`
+	Metadata          map[string]string `json:"metadata"`
+}
+
+// String colorized string message.
+func printStat(stat statMessage, maxKeyLen int) {
+	// Format properly for alignment based on maxKey length
+	stat.Key = fmt.Sprintf("%-10s: %-*.*s", "Name", maxKeyLen, maxKeyLen, stat.Key)
+	console.Println(console.Colorize("Name", stat.Key))
+	console.Println(fmt.Sprintf("%-10s: %s ", "Date", stat.Date.Format(printDate)))
+	console.Println(fmt.Sprintf("%-10s: %-6s ", "Size", humanize.IBytes(uint64(stat.Size))))
+	if stat.ETag != "" {
+		console.Println(fmt.Sprintf("%-10s: %s ", "ETag", stat.ETag))
+	}
+	console.Println(fmt.Sprintf("%-10s: %s ", "Type", stat.Type))
+
+	var maxKey = 0
+	for k := range stat.Metadata {
+		if len(k) > maxKey {
+			maxKey = len(k)
+		}
+	}
+	if len(stat.Metadata) > 0 {
+		console.Println(fmt.Sprintf("%-10s:", "Metadata"))
+		for k, v := range stat.Metadata {
+			console.Println(fmt.Sprintf("  %-*.*s: %s ", maxKey, maxKey, k, v))
+		}
+	}
+	maxKey = 0
+	for k := range stat.EncryptionHeaders {
+		if len(k) > maxKey {
+			maxKey = len(k)
+		}
+	}
+	if len(stat.EncryptionHeaders) > 0 {
+		console.Println(fmt.Sprintf("%-10s:", "Encrypted"))
+		for k, v := range stat.EncryptionHeaders {
+			console.Println(fmt.Sprintf("  %-*.*s: %s ", maxKey, maxKey, k, v))
+		}
+	}
+	console.Println()
+}
+
+// Prints all the stats.
+func printStats(stats ...statMessage) {
+	var maxKey = 0
+	for _, stat := range stats {
+		if len(stat.Key) > maxKey {
+			maxKey = len(stat.Key)
+		}
```

Comment:
I see why you wanted this so that the alignment seems to be correct for all objects, i think it is better to do it more efficiently than have a tabular column adjustment.  You can pick a maximum value and add `...` at the beginning so that they are ellipsided and we can display the file name instead of a full absolute path. 

This way we can be efficient and also never loose information.. for an expanded view user can always do `mc stat --json`.  Terminal UI can choose sane defaults for these things.